### ISSUE TITLE
Proposal: CLI App Boilerplate

### ISSUE THREAD

**Issue by graham-iheart** [Melkeydev/go-blueprint#52]
While writing CLI tools with Cobra and Viper is pretty easy, it would great if this supported setting up that boilerplate and stuff. I'd be happy to work on that contribution, but I wanted to put an issue first to gauge interest. Lmk what you think! :)

Btw great work on this dude, this is sick!

**Comment by Melkeydev** [comment]
Hey - do you mean setting up boiler plate for a CLI project? instead of an HTTP server?

**Comment by LukeTarr** [comment]
I had this same idea actually! @Melkeydev  I think @graham-iheart is suggesting that we have a prompt to setup the boilerplate for a cli app that automatically pulls in Cobra + Viper and gives some examples of how to configure those to make it your own. 

@graham-iheart let me know if you would want any help on this, I'd totally be down to work together on it! Great suggestion!

**Comment by basokant** [comment]
woah this is pretty meta... I like it!

**Comment by Melkeydev** [comment]
This is a great idea. Totally down for this if we can execute it correctly in the CLI. 
Totally open to suggestions - LMK who wants to claim it
@basokant @LukeTarr @graham-iheart

**Comment by LukeTarr** [comment]
Yep I can take a stab at it over the weekend @Melkeydev. I forked the repo over [here.](https://github.com/LukeTarr/go-blueprint). I'll do my best to keep up with the standard way of doing everything that's already in the codebase, Thanks!

**Comment by LukeTarr** [comment]
As an update to this, I'm basically done. I only brought in Cobra because I figured if people wanted to use Viper they might use it in a server app too so I don't want to confuse anyone or complicate it. I did have to do some hacky things because the cobra project type is going to have functions that the others don't have and we have to omit a bunch of others, but it does work and still maintains much of it's readability just some convention breaking (like calling a template func directly). All that's left is a for a huge merge on my part (looks like the way templates are stored has changed), so I'll bring us up to speed on that and submit a PR! Thanks!
